functions/src/modules/SE.functions.ts

```diff
@@ -88,3 +89,227 @@ export const processNewAllotment = functions.database
     });
     return 1;
   });
+
+
+const timeToMins = (time) => {
+    let hours = time.split(':')[0];
+    let mins = time.split(':')[1];
+    return (+hours * 60) + (+mins);
+}
+
+export const importSpreadSheetData = functions.runWith({
+  timeoutSeconds: 300,
+  memory: '1GB'
+}).https.onRequest( async (req, res) => {
+  const auth = await google.auth.getClient({
+      scopes: ["https://www.googleapis.com/auth/spreadsheets.readonly"]
+  });
+
+  const sheets = google.sheets({ version: "v4", auth });
+
+  const spreadsheetId = functions.config().sound_editing.spreadsheet_id;
+
+  const currentSheets = await sheets.spreadsheets.get({
+      spreadsheetId
+  });
+
+  ////////////////////////
+  // Building an index
+  ////////////////////////
+
+  const sheetTitles = [];
+  let columnsIndex = {}; // { ColumnName: column_index }
+
+  for (let i = 0; i < currentSheets.data.sheets.length; i++) {
+      let { title } = currentSheets.data.sheets[i].properties;
+      const result = await sheets.spreadsheets.values.get({
+          spreadsheetId,
+          range: `${title}!1:1` // all the cells in the first row `Column headers`
+      });
+
+      if (!result.data.values || result.data.values.length === 0) {
+          console.log(`No data found in sheet: ${title}`);
+          continue;
+      }
+  
+      // Filtering the sheets
+      const row = result.data.values[0] 
+      if (row.indexOf('Beginning') > -1 && row.indexOf('Ending') > -1) {
+          sheetTitles.push(title);
+          row.forEach((col, i) => {
+              columnsIndex[col.replace(/ /g, '')] = i;
+          });
+      }
+  }
+
+
+  let resolutions = ['ok', 'drop', 'duplicate', 'on_hold', 'reallot', 'repeat', 'derivative'];
+  let resolution, fidCheckRes;
+  let lastFile,
+      summary = { 
+          addedChunks: 0, 
+          emptyValues: 0, 
+          misplaced: 0, 
+          noResolution: 0, 
+          overlappingChunks: [],
+          modified: []
+      };
+
+  for (let i = 0; i < sheetTitles.length; i++) {
+      const result = await sheets
+          .spreadsheets.values.get({
+              spreadsheetId,
+              range: `${sheetTitles[i]}!A2:AZ1000`
```

Comment:
Isn't there another way to get all the sheet? Some lists will have more than 1000 rows… So think of another, more reliable way.

Reply:
The maximum number of rows that can be obtained in one call is 1000 ... Is it okay if I get the data in multiple calls ?

Reply:
@anantakrishna